File: 136.single-number.py
#
# @lc app=leetcode id=136 lang=python3
#
# [136] Single Number
#

# @lc code=start
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        # use XOR operation since two identical number XOR together will be 0
        # and any number XOR 0 will remain the same
        ans = nums[0]
        for i in range(1, len(nums)):
            n = nums[i]
            ans = ans^n
        return ans
        
# @lc code=end

File: 136.single-number.go
/*
 * @lc app=leetcode id=136 lang=golang
 *
 * [136] Single Number
 */

// @lc code=start
func singleNumber(nums []int) int {
	ans := 0
	for _, val := range nums {
		ans = ans ^ val
	}
	return ans

}

// @lc code=end

